### DISCUSSION TITLE
Get the appId in cloud function

### DISCUSSION THREAD

**Discussion by tristan-garaud** [firebase/firebase-js-sdk#7204] [Q&A]
Hi,
I want to know from which client the user is signing in.
I would like to know if it's possible to get the appId or something else that allows me to know where he is coming from when the user is signing inside my cloud function ?
Thanks

**Comment by ryanmio** [comment]
You can identify the client source in several ways when working with Cloud Functions:

1. **HTTP Headers**: Include a custom HTTP header like `X-App-Id` when making the request to your Cloud Function. In the function, you can retrieve it as follows:

    ```javascript
    exports.yourCloudFunction = functions.https.onRequest((req, res) => {
      const appId = req.get('X-App-Id');
      // Do something with appId
    });
    ```

2. **Token Payload**: If using Firebase Authentication, you might include `appId` in the token payload. This would allow your function to read it after decoding the token.

3. **Function Arguments**: Directly pass the `appId` as an argument to the Cloud Function.

Choose the method that best fits your use case to identify the client.

Hope this helps!
